client/client.go

```diff
@@ -431,9 +441,24 @@ type MetricsTimer = metrics.Timer
 // MetricsNopHandler is a noop handler that does nothing with the metrics.
 var MetricsNopHandler = metrics.NopHandler
 
+// Dial creates an instance of a workflow client. This will attempt to connect
+// to the server eagerly and will return an error if the server is not
+// available.
+func Dial(options Options) (Client, error) {
+	return internal.DialClient(options)
```

Comment:
```suggestion
	return internal.Dial(options)
```
?

Reply:
Just `Dial` in a `client` package makes sense, but just `Dial` in an internal package where you could be dialing other things does not. `worker.New` -> `internal.NewWorker`, `client.Options` -> `internal.ClientOptions`, etc, etc.